### PR TITLE
[EPMCDLAB-1761]: Added endpoint for getting all user's environment

### PR DESCRIPTION
**Pull request by ochaparin**

### CODE REVIEW

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import java.util.Collections;
+import java.util.List;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AdminPageResourceInfo {
+
+	@JsonProperty("type")
+	private String resourceType;
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java]: New enum should be introduced : NOTEBOOK, EDGE

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+	@JsonProperty("resource_id/name")
+	private String resourceIdName;
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java]: resource name not needed on UI

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+	@JsonProperty("status")
+	private String resourceStatus;
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java]: UserInstanceStatus should be used

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+
+import java.util.Collections;
+import java.util.List;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AdminPageResourceInfo {
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java]: rename to UserResourceInfo

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/EnvironmentResource.java [hunk 5]
```diff
@@ -122,6 +130,16 @@ public Response stopCluster(@Auth UserInfo userInfo, @NotEmpty String user,
 		return Response.ok().build();
 	}
 
+	@POST
+	@Consumes(MediaType.TEXT_PLAIN)
+	@Produces(MediaType.APPLICATION_JSON)
+	@Path("terminate/edge")
+	public Response terminateEdge(@Auth UserInfo userInfo, @NotEmpty String user) {
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/resources/EnvironmentResource.java]: such endpoint not needed

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java (hunk 4 above)
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java]: Done

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java (hunk 1 above)
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java]: Added

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java (hunk 2 above)
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java]: Removed 'resource_id' for Edge node

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java (hunk 3 above)
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/resources/dto/AdminPageResourceInfo.java]: Refactored

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/EnvironmentResource.java (hunk 5 above)
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/resources/EnvironmentResource.java]: Removed

**Pull request merged**